### ISSUE TITLE
RCS1261 shouldn't trigger on `IDbContextFactory<TContext>.CreateDbContext()`

### ISSUE THREAD

**Issue by Sella-GH** [JosefPihrt/Roslynator#1760]
### Description

As per https://github.com/dotnet/roslyn-analyzers/issues/7036 this Analyzer shouldn't trigger on `IDbContextFactory<TContext>.CreateDbContext()` due to the Async version was only added for specific edge-case purposes and not for general use.

### Reproduction Steps

Add `using AppDbContext dbContext = _dbContextFactory.CreateDbContext();` in an asynchronous method.

### Other information

Some more insights can be found in https://github.com/dotnet/efcore/issues/26630

**Comment by josefpihrt** [comment]
Thanks for the report and the links.

RCS1261 is **not** the same rule as CA1849. CA1849 ("call async methods when in an async method") is what [roslyn-analyzers#7036](https://github.com/dotnet/roslyn-analyzers/issues/7036) and [efcore#26630](https://github.com/dotnet/efcore/issues/26630) discuss - that rule targets replacing synchronous **creation** calls (e.g. `CreateDbContext()` → `CreateDbContextAsync()`).

RCS1261 ("Resource can be disposed asynchronously") targets **disposal**. It reports on the `using` keyword when the resource type implements `IAsyncDisposable`. It does not inspect whether the initializer came from a sync or async factory method.

For your scenario (already-`async` method):

```csharp
// before
using AppDbContext dbContext = _dbContextFactory.CreateDbContext();

// after applying the RCS1261 code fix
await using AppDbContext dbContext = _dbContextFactory.CreateDbContext();
```

`CreateDbContext()` is left unchanged. The fix only adds `await` to `using` so disposal goes through `DisposeAsync()`, which is a supported EF Core API.

Note: if the containing method is *not* already `async` but returns `Task`/`Task<T>`, the code fix also adds the `async` modifier and `await`s return expressions in that method. It still does not rewrite creation calls to async counterparts.

If you prefer not to use `await using` for DbContext, you can suppress this rule:

```ini
dotnet_diagnostic.RCS1261.severity = none
```

Closing as by-design. The linked CA1849 discussion is about a different diagnostic.

**Comment by Sella-GH** [comment]
Thanks for the detailed explanation!

Then I misunderstood and falsely interpreted the Analyzer claims. It does make more sense now after you have provided the insights.
